### PR TITLE
Fix TypeError: unsupported operand type(s) for -: 'NoneType' and 'int' on module ec2_asg

### PR DESCRIPTION
**Pull request by galindro**
This fixes the issue: https://github.com/ansible/ansible-modules-core/issues/2877

### CODE REVIEW

File: cloud/amazon/ec2_asg.py [hunk 1]
```diff
@@ -555,6 +555,15 @@ def replace(connection, module):
...
     as_group = connection.get_all_groups(names=[group_name])[0]
+
+    #check if min_size/max_size/desired capacity have been specified and if not use ASG values
+    if min_size is None:
+        min_size = as_group.min_size
+    if max_size is None:
+        max_size = as_group.max_size
+    if desired_capacity is None:
+        desired_capacity = as_group.desired_capacity
+
```
Comment (bcoca) [inline cloud/amazon/ec2_asg.py:566]: Why not make these a 2nd parameter for the .get()s above?

**Comment by gregdek** [comment]
Thanks @galindro for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

**Comment by shaharmor** [comment]
@gregdek How long exactly? Its been nearly 10 months already...

File: cloud/amazon/ec2_asg.py [hunk 2]
```diff
@@ -672,6 +674,15 @@ def terminate_batch(connection, module, replace_instances, initial_instances, le
     break_loop = False
 
     as_group = connection.get_all_groups(names=[group_name])[0]
+
+    #check if min_size/max_size/desired capacity have been specified and if not use ASG values
+    if min_size is None:
+        min_size = as_group.min_size
+    if max_size is None:
```
Comment (shaharmor) [inline cloud/amazon/ec2_asg.py:681]: There is no `max_size` variable here

**Comment by gregdek** [comment]
You might consider attending the Core team IRC meetings to ask for expedited review:

https://github.com/ansible/community/blob/master/MEETINGS.md

**Comment by gregdek** [comment]
Thanks @galindro for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@galindro A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**